src/components/MenuItem.tsx

```diff
@@ -253,6 +254,18 @@ type MenuItemBaseProps = {
 
     /** Handles what to do when the item is focused */
     onFocus?: () => void;
+
+    /** Whether to show the tooltip */
+    shouldRenderTooltip?: boolean;
+
+    /** Whether to align the tooltip left */
+    shouldForceRenderingTooltipLeft?: boolean;
+
+    /** Additional styles for tooltip wrapper */
+    tooltipWrapperStyle?: StyleProp<ViewStyle>;
+
+    /** Render custom content inside the tooltip. */
+    renderTooltipContent?: () => ReactNode;
```

Comment:
My observation: in this component, in the context of the typing I'm quoting, `shouldRenderTooltip` is truthy **if and only if** `renderTooltipContent` is defined.

This is not a well-designed data structure. Unnecessary and dangerous combinations are possible (like `shouldRenderTooltip: true` and `renderTooltipContent: undefined`. Additionally, once we fix this, it's likely that we'll open a path for simplifying `EducationalTooltip`, but that's just another argument.